RAGchain/reranker/bm25.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+from rank_bm25 import BM25Okapi
+from transformers import AutoTokenizer
+
+from RAGchain.reranker.base import BaseReranker
+from RAGchain.schema import Passage
+
+
+class BM25Reranker(BaseReranker):
+    '''
+    BM25Reranker is a reranker based on BM25.
+    You can rerank the passages with the instruction using BM25Reranker.
+    '''
+
+    def __init__(self, save_path: str, tokenizer_name: str = "gpt2", *args, **kwargs):
+        self.save_path = save_path
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+    def rerank(self, query: str, passages: List[Passage]) -> List[Passage]:
+        contents: List[str] = [passage.content for passage in passages]
+
+        bm25 = BM25Okapi(contents)
```

Comment:
Tokenize contents before putting to BM25. Please look BM25retrieval, which tokenizes all contents using AutoTokenizer at ingest() method, and get tokenized tokens from .pkl file.